File: ChickInRun/Assets/SubModel/san_lava/sam_runaway.js
#pragma strict

private var wayPoint :GameObject;
private var wayPointPos : Vector3;
//private var wayPointRot : Quaternion;

private var runawayPos : Vector3;
private var timer : float = 1.0f;


 
 

 private var speed : float= 3.0;
 private var MaxDistance: float =2.0;
 private var MinDistance: float =0.5;
 
 
 function Start ()
 {
      wayPoint = GameObject.Find("wayPoint");
 }
 
 function Update ()
 {

 	
 	//wayPointPos = new Vector3(wayPoint.transform.position.x, transform.position.y, wayPoint.transform.position.z);
 	//wayPointRot = Quaternion.LookRotation (-wayPointPos, Vector3.up);
 	runawayPos=transform.position - wayPoint.transform.position;
 	//runawayPos.z = 0.0f;


 	
 	if(Vector3.Distance(transform.position,wayPoint.transform.position)<MaxDistance){
 		  	if(timer > 0)
    		  {
         		  timer -= Time.deltaTime;
     		  }
 	
	      
	      //weasel will follow the waypoint.
	      transform.position = Vector3.MoveTowards(transform.position, runawayPos*MinDistance, speed * Time.deltaTime);
	       GetComponent.<Animation>().Play("mannyonsam_Walk");
          
          transform.LookAt(-wayPointPos);

 	}
 	
 	if(timer <= 0)
     	 {
     	 	stop();
     	 	timer = 1.0f;
     	 
     	 }
 	
 }
 
 function stop(){
 
 	yield WaitForSeconds(2.0);
 }

File: ChickInRun/Assets/Script/samRunaway.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class samRunaway : MonoBehaviour
{
    private GameObject wayPoint;
    private Vector3 wayPointPos;
    private Vector3 runawayPos;
    private float timer = 1.0f;

    private float speed = 3.0f;
    private float maxDistance = 2.0f;
    private float minDistance = 0.5f;

    // Start is called before the first frame update
    void Start()
    {
        wayPoint = GameObject.Find("wayPoint");
    }

    // Update is called once per frame
    void Update()
    {
        wayPointPos = new Vector3(wayPoint.transform.position.x, wayPoint.transform.position.y, wayPoint.transform.position.z);
        runawayPos = transform.position - wayPoint.transform.position;

        if (Vector3.Distance(transform.position, wayPoint.transform.position) < maxDistance)
        {
            if(timer > 0)
            {
                timer -= Time.deltaTime;
            }

            transform.position = Vector3.MoveTowards(transform.position, runawayPos * minDistance, speed * Time.deltaTime);
            GetComponent<Animation>().Play("mannyonsam_Walk");
            transform.LookAt(-wayPointPos);
        }
        if(timer <=0)
        {
            StartCoroutine("StopSam");
            timer = 1.0f;
        }
    }

    private IEnumerator StopSam()
    {
        yield return new WaitForSeconds(2.0f);
    }
}
